types/stellar-sdk/index.d.ts

```diff
@@ -876,4 +879,9 @@ export namespace xdr {
     class Asset extends XDRStruct { }
     class Memo extends XDRStruct { }
     class TransactionEnvelope extends XDRStruct { }
+    class DecoratedSignature extends XDRStruct {
+      constructor(keys: { hint: SignatureHint, signature: Signature })
```

Comment:
Could you add the types:
`hint(): Buffer`
`signature(): Buffer`

I think...

Reply:
What do you mean?

Reply:
Those methods at least used to exist (I work on a fork). Those two functions on the DecoratedSignature can be used to do something like this to see who has signed a transaction envelope:
```js
  const transactionSignatures = transaction.signatures.map((sig: any) => sig.signature())

  return signers.filter(kp => transactionSignatures.some(sig => kp.verify(transaction.hash(), sig)))
```

Reply:
I just reviewed the source code files again and still didn't find those. I can add them if you want tho.

```
   struct DecoratedSignature
   {
       SignatureHint hint;  // last 4 bytes of the public key, used as a hint
       Signature signature; // actual signature
   };
```

Reply:
That is the source code that defines the functions I described above. XDR -> JS turns them into getters. Disappointing to not have added it in when you were just there.